### PR TITLE
Make use of IntoGStr/IntoStrV

### LINKED ISSUE #1240: Use `IntoGStr` / `IntoStrV` and collection types more widely in manual code
See also https://github.com/gtk-rs/gtk-rs-core/issues/884

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
Fixes #1240

### CODE REVIEW

File: gtk4/src/param_spec_expression.rs [hunk 1]
```diff
@@ -84,15 +86,26 @@ impl FromGlibPtrFull<*mut gobject_ffi::GParamSpec> for ParamSpecExpression {
 impl ParamSpecExpression {
     #[allow(clippy::new_ret_no_self)]
     #[doc(alias = "gtk_param_spec_expression")]
-    pub fn new(name: &str, nick: &str, blurb: &str, flags: glib::ParamFlags) -> ParamSpec {
+    pub fn new(
+        name: impl IntoGStr,
+        nick: impl IntoGStr,
+        blurb: impl IntoGStr,
```
Comment (sdroege) [inline gtk4/src/param_spec_expression.rs:92]: These are optional or not? `IntoOptionalGStr`

File: gtk4/src/constraint_layout.rs [hunk 2]
```diff
@@ -10,7 +10,7 @@ impl ConstraintLayout {
     #[doc(alias = "add_constraints_from_descriptionv")]
     pub fn add_constraints_from_description<W: IsA<Widget>>(
         &self,
-        lines: &[&str],
+        lines: impl IntoStrV,
         hspacing: i32,
         vspacing: i32,
         views: &HashMap<&str, &W>,
```
Comment (sdroege) [inline gtk4/src/constraint_layout.rs:16]: Could also make this a `IntoIterator<Item = (&str, &W)>` FWIW

**Review by sdroege — COMMENTED** [review]
Looks good to me otherwise. You could also look at `&str` and `Vec<String>` / `Vec<GString>` return values

File: gtk4/src/param_spec_expression.rs (hunk 1 above)
Comment (bilelmoussaoui) [reply gtk4/src/param_spec_expression.rs:92]: Yes they are optional

**Comment by bilelmoussaoui** [comment]
> Looks good to me otherwise. You could also look at `&str` and `Vec<String>` / `Vec<GString>` return values

Checked those, there were not that many. Only one case where we were returning `String` instead of `GString`

**Comment by bilelmoussaoui** [comment]
> > Looks good to me otherwise. You could also look at `&str` and `Vec<String>` / `Vec<GString>` return values
> 
> Checked those, there were not that many. Only one case where we were returning `String` instead of `GString`

actually, i just noticed that i forgot doing the same thing for subclassing code. Will handle that first

**Pull request merged**